### PR TITLE
Split the Nassau SoupBinTCP Performance Test into a client and server

### PR DESCRIPTION
**Pull request by JanStureNielsen**
As noted in #53, split the performance test into a client and server.

### CODE REVIEW

File: examples/soupbintcp-client/README.md [hunk 1]
```diff
@@ -15,20 +15,20 @@ an Unsequenced Data packet and receiving an Sequenced Data packet.
...
-   99.90%:     107.52 µs
-   99.99%:     233.09 µs
-  100.00%:    1909.76 µs
+      Min:      11.18 µs
+   50.00%:      12.65 µs
+   90.00%:      13.49 µs
+   99.00%:      24.45 µs
+   99.90%:      30.67 µs
+   99.99%:      38.50 µs
+  100.00%:      75.84 µs
```
Comment (chrjohn) [inline examples/soupbintcp-client/README.md]: Sorry, just a short question: was there a specific change done that lead to this massive improvement or is this just new hardware? ;) It is not stated which hardware was used (or I missed it).

Thanks,
Chris.

Comment (jvirtanen) [reply examples/soupbintcp-client/README.md]: I think the hardware (and the operating system?) might have changed.

For this reason on `master` the performance test [has been run on AWS c4.xlarge](https://github.com/paritytrading/nassau/tree/master/tests/soupbintcp-perf-test) in order to make it easy to reproduce the test results.

Comment (JanStureNielsen) [reply examples/soupbintcp-client/README.md]: As @jvirtanen noted, it is purely hardware -- the first one was an (old) MacBook Air, the second on my Linux development machine.

Comment (JanStureNielsen) [reply examples/soupbintcp-client/README.md]: The development metrics have now been reverted to the original.

File: examples/soupbintcp-client/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Nassau SoupBinTCP Test Client
+
+Nassau SoupBinTCP Test Client is an example application implementing SoupBinTCP.
+
+## Usage
+
+Run Nassau SoupBinTCP Test Client with Java:
+
+```
+java -jar nassau-soupbintcp-perf-test.jar <host> <port> <packets>
```
Comment (jvirtanen) [inline examples/soupbintcp-client/README.md]: This should be updated:
```
java -jar nassau-soupbintcp-client.jar <host> <port> <packets>
```

File: examples/soupbintcp-client/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Nassau SoupBinTCP Test Client
+
+Nassau SoupBinTCP Test Client is an example application implementing SoupBinTCP.
```
Comment (jvirtanen) [inline examples/soupbintcp-client/README.md]: Maybe the end could be something like this:
> ... implementing a SoupBinTCP client.

Likewise on the server side.

File: examples/soupbintcp-client/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+Run Nassau SoupBinTCP Test Client with Java:
+
+```
+java -jar nassau-soupbintcp-perf-test.jar <host> <port> <packets>
+```
+
+The application measures the round-trip time (RTT) between sending
+an Unsequenced Data packet and receiving an Sequenced Data packet.
+
+## Results
```
Comment (jvirtanen) [inline examples/soupbintcp-client/README.md]: An empty line should follow heading. 🙂

File: examples/soupbintcp-client/README.md (hunk 2 above)
Comment (JanStureNielsen) [reply examples/soupbintcp-client/README.md]: Whoops -- done.

File: examples/soupbintcp-client/README.md (hunk 3 above)
Comment (JanStureNielsen) [reply examples/soupbintcp-client/README.md]: Done

File: examples/soupbintcp-client/README.md (hunk 4 above)
Comment (JanStureNielsen) [reply examples/soupbintcp-client/README.md]: Done

File: examples/soupbintcp-client/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+Results (n = 50000):
+
+      Min:       4.79 µs
+   50.00%:       5.43 µs
+   90.00%:       6.86 µs
+   99.00%:       8.72 µs
+   99.90%:      13.70 µs
+   99.99%:      28.93 µs
+  100.00%:      45.15 µs
+
```
Comment (jvirtanen) [inline examples/soupbintcp-client/README.md]: I just noticed that this code block is not closed. It should be like this:

       ...          ...
       99.99%:      28.93 µs
      100.00%:      45.15 µs
    ```

File: examples/soupbintcp-server/pom.xml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  <dependencies>
+    <dependency>
+      <groupId>${project.groupId}</groupId>
+      <artifactId>nassau-core</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.hdrhistogram</groupId>
+      <artifactId>HdrHistogram</artifactId>
+    </dependency>
```
Comment (jvirtanen) [inline examples/soupbintcp-server/pom.xml]: I thought I mentioned about this earlier, but it looks like I forgot: this library is actually not used on the server side, so this dependency can be removed.

File: scripts/archive.txt [hunk 7]
```diff
@@ -12,5 +12,7 @@ libraries/core/README.md
 libraries/util/README.md
 tests/binaryfile-perf-test/README.md
 tests/binaryfile-perf-test/nassau-binaryfile-perf-test.jar
-tests/soupbintcp-perf-test/README.md
-tests/soupbintcp-perf-test/nassau-soupbintcp-perf-test.jar
+examples/soupbintcp-client/README.md
+examples/soupbintcp-client/nassau-soupbintcp-client.jar
+examples/soupbintcp-server/README.md
+examples/soupbintcp-server/nassau-soupbintcp-server.jar
```
Comment (jvirtanen) [inline scripts/archive.txt]: A nit: let's keep these in the alphabetical order. 🤓

Comment (JanStureNielsen) [reply scripts/archive.txt]: Done

File: examples/soupbintcp-client/README.md (hunk 5 above)
Comment (JanStureNielsen) [reply examples/soupbintcp-client/README.md]: Whoops - done.

File: examples/soupbintcp-server/pom.xml (hunk 6 above)
Comment (JanStureNielsen) [reply examples/soupbintcp-server/pom.xml]: Done

File: examples/soupbintcp-client/README.md (hunk 5 above)
Comment (JanStureNielsen) [reply examples/soupbintcp-client/README.md]: Done

**Comment by JanStureNielsen** [comment]
Thanks for the review, @jvirtanen -- all comments are now resolved.

**Comment by jvirtanen** [comment]
Thank you, @JanStureNielsen! 🙌

**Pull request merged**